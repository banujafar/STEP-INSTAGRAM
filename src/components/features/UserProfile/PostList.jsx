import { AiFillHeart, AiOutlineHeart } from "react-icons/ai";
import { FaComment, FaRegComment } from "react-icons/fa";
import { useSelector } from "react-redux";
import { Link } from "react-router-dom";
import PostCard from "./PostCard";

const PostList = () => {
  const userInfo = useSelector((state) => state.userProfile.userInfo);
  const posts = userInfo.posts;
  console.log(posts);
  return (
    <div className="post-list flex flex-wrap w-full justify-start max-w-4xl mx-auto">
      {posts?.map((post, index) => (
        <PostCard key={index} post={post} />
      ))}
    </div>
  );
};
export default PostList;
