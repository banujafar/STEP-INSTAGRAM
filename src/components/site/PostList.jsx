import { BsFillFilePersonFill, BsPersonCircle } from "react-icons/bs";
import PostCard from "./PostCard";
import { AiOutlineInsertRowAbove } from "react-icons/ai";
import React, { useState } from "react";

const PostList = ({ userData }) => {
  const posts = userData.posts;
  const copiedPosts = posts ? [...posts] : [];
  copiedPosts.sort((a, b) => b.timestamp - a.timestamp);
  const [isActive, setIsActive] = useState("POSTS");
  const handleChange = (option) => {
    setIsActive(option);
  };

  return (
    <div className="post-list flex flex-wrap w-full justify-start max-w-4xl mx-auto z-10 border-t-2 relative">
      <ul className="flex absolute top-0 left-96 mt-3">
        <li
          className={`mr-12 flex gap-1 justify-center items-center cursor-pointer ${
            isActive === "POSTS" ? "text-black font-bold" : ""
          }`}
          onClick={() => handleChange("POSTS")}
        >
          <AiOutlineInsertRowAbove size={20} />
          POSTS
        </li>
        <li
          className={`flex gap-1 justify-center items-center cursor-pointer ${
            isActive === "TAGGED" ? "text-black font-bold" : ""
          }`}
          onClick={() => handleChange("TAGGED")}
        >
          <BsFillFilePersonFill />
          TAGGED
        </li>
      </ul>
      {isActive === "POSTS" ? (
        !!copiedPosts.length ? (
          copiedPosts?.map((post, index) => (
            <PostCard key={index} post={post} data-testid="post-card" />
          ))
        ) : (
          <div className="w-full h-60 mt-12 flex flex-col items-center justify-center ml-8 text-2xl">
            <span className="font-bold">No Posts Yet</span>
          </div>
        )
      ) : (
        <div className="w-full h-60 mt-12 flex flex-col items-center justify-center ml-8 ">
          <BsPersonCircle size={32} />
          <span className="font-bold mt-2">No Photos</span>
        </div>
      )}
    </div>
  );
};
export default PostList;
