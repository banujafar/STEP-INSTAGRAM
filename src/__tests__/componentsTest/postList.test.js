import "whatwg-fetch";
import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import PostList from "../../components/site/PostList";
describe("render Posts List component", () => {
  let userData = {
    username: "banu123",
    firstName: "Banu",
    lastName: "Ceferli",
    posts: [
      {
        authorUsername: "alexbird",
        caption: "Photo was made from scanned 35mm film",
        postId: "02c6f4fdefff4409b12b793cfb5e0eeb",
        timestamp: 1693024855,
        likes: [
          { authorUsername: "kate_oliver" },
          { authorUsername: "diasik" },
        ],
      },
      {
        authorUsername: "alexbird",
        caption: "",
        postId: "d87711dd37004d88af06dee100d9c20a",
        timestamp: 1693038987,
        likes: [
          { authorUsername: "kate_oliver" },
          { authorUsername: "diasik" },
        ],
      },
    ],
    subscriptions: [],
    subscribers: [],
  };

  it("when click list change header", async () => {
    render(<PostList userData={userData} />);
    expect(screen.getAllByText(2)).toHaveLength(2);
    fireEvent.click(screen.getByText(/TAGGED/i));
    expect(screen.getByText("No Photos")).toBeInTheDocument();
  });
  
  it("snapshot testing for PostList component", () => {
    const tree = render(<PostList userData={userData} />);
    expect(tree).toMatchSnapshot();
  });
});
