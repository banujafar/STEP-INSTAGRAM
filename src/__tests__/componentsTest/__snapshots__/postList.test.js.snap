// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Posts List component snapshot testing for PostList component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="post-list flex flex-wrap w-full justify-start max-w-4xl mx-auto z-10 border-t-2 relative"
      >
        <ul
          class="flex absolute top-0 left-96 mt-3"
        >
          <li
            class="mr-12 flex gap-1 justify-center items-center cursor-pointer text-black font-bold"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              t="1569683507961"
              version="1.1"
              viewBox="0 0 1024 1024"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs />
              <path
                d="M878.7 336H145.3c-18.4 0-33.3 14.3-33.3 32v464c0 17.7 14.9 32 33.3 32h733.3c18.4 0 33.3-14.3 33.3-32V368c0.1-17.7-14.8-32-33.2-32zM360 792H184V632h176v160z m0-224H184V408h176v160z m240 224H424V632h176v160z m0-224H424V408h176v160z m240 224H664V632h176v160z m0-224H664V408h176v160zM904 160H120c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8z"
              />
            </svg>
            POSTS
          </li>
          <li
            class="flex gap-1 justify-center items-center cursor-pointer "
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0H4a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h8a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2zm-1 7a3 3 0 1 1-6 0 3 3 0 0 1 6 0zm-3 4c2.623 0 4.146.826 5 1.755V14a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1v-1.245C3.854 11.825 5.377 11 8 11z"
              />
            </svg>
            TAGGED
          </li>
        </ul>
        <div
          class="w-72 h-60 mr-2.5 mt-12"
          data-testid="post-modal"
        >
          <div
            class="w-full h-full relative group/item shadow-lg cursor-pointer"
          >
            <img
              alt="Post"
              class="w-full h-full"
            />
            <div
              class="bg-opacity-0 hover:bg-black-rgba transition-all duration-500 w-full h-full absolute left-0 top-0"
            >
              <span
                class="group/edit invisible group-hover/item:visible absolute left-1/3 top-1/2 flex"
              >
                <div
                  class="flex justify-start flex-col"
                >
                  <div
                    class="flex items-center space-x-4"
                  >
                    <button
                      class="flex items-center text-gray-600"
                    >
                      <svg
                        class="text-2xl"
                        fill="white"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 1024 1024"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
                        />
                      </svg>
                      <span
                        class="text-white ml-2"
                      >
                        2
                      </span>
                    </button>
                    <button
                      class="flex items-center text-gray-600"
                    >
                      <svg
                        class="text-2xl"
                        fill="white"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 512 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          class="w-72 h-60 mr-2.5 mt-12"
          data-testid="post-modal"
        >
          <div
            class="w-full h-full relative group/item shadow-lg cursor-pointer"
          >
            <img
              alt="Post"
              class="w-full h-full"
            />
            <div
              class="bg-opacity-0 hover:bg-black-rgba transition-all duration-500 w-full h-full absolute left-0 top-0"
            >
              <span
                class="group/edit invisible group-hover/item:visible absolute left-1/3 top-1/2 flex"
              >
                <div
                  class="flex justify-start flex-col"
                >
                  <div
                    class="flex items-center space-x-4"
                  >
                    <button
                      class="flex items-center text-gray-600"
                    >
                      <svg
                        class="text-2xl"
                        fill="white"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 1024 1024"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
                        />
                      </svg>
                      <span
                        class="text-white ml-2"
                      >
                        2
                      </span>
                    </button>
                    <button
                      class="flex items-center text-gray-600"
                    >
                      <svg
                        class="text-2xl"
                        fill="white"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 512 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="post-list flex flex-wrap w-full justify-start max-w-4xl mx-auto z-10 border-t-2 relative"
    >
      <ul
        class="flex absolute top-0 left-96 mt-3"
      >
        <li
          class="mr-12 flex gap-1 justify-center items-center cursor-pointer text-black font-bold"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            t="1569683507961"
            version="1.1"
            viewBox="0 0 1024 1024"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <path
              d="M878.7 336H145.3c-18.4 0-33.3 14.3-33.3 32v464c0 17.7 14.9 32 33.3 32h733.3c18.4 0 33.3-14.3 33.3-32V368c0.1-17.7-14.8-32-33.2-32zM360 792H184V632h176v160z m0-224H184V408h176v160z m240 224H424V632h176v160z m0-224H424V408h176v160z m240 224H664V632h176v160z m0-224H664V408h176v160zM904 160H120c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8z"
            />
          </svg>
          POSTS
        </li>
        <li
          class="flex gap-1 justify-center items-center cursor-pointer "
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0H4a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h8a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2zm-1 7a3 3 0 1 1-6 0 3 3 0 0 1 6 0zm-3 4c2.623 0 4.146.826 5 1.755V14a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1v-1.245C3.854 11.825 5.377 11 8 11z"
            />
          </svg>
          TAGGED
        </li>
      </ul>
      <div
        class="w-72 h-60 mr-2.5 mt-12"
        data-testid="post-modal"
      >
        <div
          class="w-full h-full relative group/item shadow-lg cursor-pointer"
        >
          <img
            alt="Post"
            class="w-full h-full"
          />
          <div
            class="bg-opacity-0 hover:bg-black-rgba transition-all duration-500 w-full h-full absolute left-0 top-0"
          >
            <span
              class="group/edit invisible group-hover/item:visible absolute left-1/3 top-1/2 flex"
            >
              <div
                class="flex justify-start flex-col"
              >
                <div
                  class="flex items-center space-x-4"
                >
                  <button
                    class="flex items-center text-gray-600"
                  >
                    <svg
                      class="text-2xl"
                      fill="white"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
                      />
                    </svg>
                    <span
                      class="text-white ml-2"
                    >
                      2
                    </span>
                  </button>
                  <button
                    class="flex items-center text-gray-600"
                  >
                    <svg
                      class="text-2xl"
                      fill="white"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </span>
          </div>
        </div>
      </div>
      <div
        class="w-72 h-60 mr-2.5 mt-12"
        data-testid="post-modal"
      >
        <div
          class="w-full h-full relative group/item shadow-lg cursor-pointer"
        >
          <img
            alt="Post"
            class="w-full h-full"
          />
          <div
            class="bg-opacity-0 hover:bg-black-rgba transition-all duration-500 w-full h-full absolute left-0 top-0"
          >
            <span
              class="group/edit invisible group-hover/item:visible absolute left-1/3 top-1/2 flex"
            >
              <div
                class="flex justify-start flex-col"
              >
                <div
                  class="flex items-center space-x-4"
                >
                  <button
                    class="flex items-center text-gray-600"
                  >
                    <svg
                      class="text-2xl"
                      fill="white"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
                      />
                    </svg>
                    <span
                      class="text-white ml-2"
                    >
                      2
                    </span>
                  </button>
                  <button
                    class="flex items-center text-gray-600"
                  >
                    <svg
                      class="text-2xl"
                      fill="white"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
